### PR TITLE
IAM authentication support

### PR DESCRIPTION
**Pull request by christiancompton**
This PR adds logic for IAM authentication using IAM credentials and the new initializer from the Watson SDK for these credential sets.  It keeps backwards compatibility with the previous authentication paradigm, which still need to be supported for users in certain availability zones, as well as users who are trying to connect existing service instances with the old credentials.

I will coordinate with @sanjeevghimire for testing & merging.

### CODE REVIEW

**Comment by christiancompton** [comment]
@sanjeevghimire  - Given that the authentication changes here did work locally, are you okay with merging it in, or do you need a full end-to-end test with downloading the model to proceed?

Edit: Tagged wrong Sanjeev!

**Comment by sanjeevghimire** [comment]
@christiancompton I am testing another round for the workshop. once that works, I will merge it.

File: ios_swift/ResumeARStarter/ViewController.swift [hunk 1]
```diff
@@ -151,16 +151,29 @@ class ViewController: UIViewController, ARSCNViewDelegate {
...
 
-        self.visualRecognition = VisualRecognition.init(apiKey: vrApiKey, version: self.VERSION)
+        // Set the Watson credentials for Visual Recognition service from the BMSCredentials.plist
+        // If using IAM authentication
+        if let apiKey = credentials["visualrecognitionApikey"] as? String {
+
+            // Create service sdks
+            self.visualRecognition = VisualRecognition(version: self.VERSION, apiKey: apiKey)
+
+        // Else for user/pwd authentication
```
Comment (stevemar) [inline ios_swift/ResumeARStarter/ViewController.swift]: you write user/pwd here but on L172 the api key is still being passed in. IIRC VizRec didn't have a user/pwd combo, wasn't it just the key that changed? `visualrecognitionApi_key` vs `visualrecognitionApikey` ? if that's the case then make the comment correct.

**Comment by christiancompton** [comment]
@stevemart @sanjeevghimire The comment above has been addressed.

**Review by stevemar — APPROVED** [review]
bueno

**Pull request merged**